src/HTMLGenerator/converter.py

```diff
@@ -14,40 +15,56 @@ def _get_path_relative_to_username(filename):
     filename_path = Path(filename)
     return str(filename_path.relative_to(s.expanduser()))
 
-def convert(top_level_output_directory, top_level_input_directory, json_with_filepaths, path_to_current_working_directory = Path.cwd()):    
+def write_configuration_file(json_with_filepaths, top_level_output_directory = Path(""), top_level_input_directory = Path(""), path_to_current_working_directory = Path.cwd()):
+    current_configuration_absolute_path = Path(path_to_current_working_directory / CURRENT_CONFIGURATION_PATH)
+    # Writing the top, preload, input and output path to the currentConfiguration.txt file.
+    try:
+        with open(current_configuration_absolute_path, 'w') as current_configuration_file:
+            if "topPath" in json_with_filepaths:
+                topDirectoryPath = json_with_filepaths["topPath"]
+            else:
+                raise Exception("topPath cannot be null/empty")
+            for configuration in json_with_filepaths["configurations"]:
+                current_configuration_file.write(
+                '/top:' + str(top_level_output_directory / topDirectoryPath) + "\n")
+                if "preloadedFiles" in configuration:
+                    current_configuration_file.write(
+                    '/preload:' + str(Path(configuration["preloadedFiles"])) + "\n")
+                if "inputPath" in configuration:
+                    current_configuration_file.write(
+                    '/input:' + str(top_level_input_directory / configuration["inputPath"]) + "\n")
+                else:
+                    raise Exception("inputPath cannot be null/empty")
+                if "outputPath" in configuration:    
+                    current_configuration_file.write(
+                    '/out:' + str(Path(configuration["outputPath"])) + "\n")
+                current_configuration_file.flush()
+                current_configuration_file.close()
+                convert()
```

Comment:
If this function is to "write_configuration_file", then it should just take care of writing the configuration file, not do the additional work of calling `convert()` as well. But since we cannot separately call `convert()`, it would be cleaner to rename the function to actually represent what it's doing. You can rename _this_ function to `convert()` and the function that's currently called `convert()` can instead be renamed to `launch_converter()`.

Reply:
So, basically, the converter.py script will expose two public methods.
1. convert(json_path)
1. run_converter()

convert(json_path):
    # Calls into write_config_file_and_run_converter()
    # This can be used by anyone who wants to run the converter through CLI

run_converter():
    # Launches LabVIEW to run the VI that converts whatever is written into the config file. This function does not need to care about who wrote the config file, it should just run under the assumption that a config file exists.
    # This can be used by the anyone who has finished creating the config file. This can be called either from this same python script (from write_config_file_and_run_converter()) or from a different python script (sequencer.py after it writes the config file independent of converter.py).

The sequencer on the other hand could do something like this:
foreach testcase:
    write_config_file() # This is specific to the sequencer, it should not use the same method present in converter.py
    run_converter() # Call into the public method of converter.py
    # Continue with the rest of the code, this should remain unchanged.

This way, we maintain the abstraction of the converter not being aware of the sequencer, and we don't need to copy-paste code from the converter in order to run the test suite. What are your thoughts? @abahetik

Reply:
Yes that is what I wanted to know. I will run_converter for the test suite by, using the function in converter.py.